### DISCUSSION TITLE
Question Regarding Parameterizations for Attenuation

### DISCUSSION THREAD

**Discussion by bvanderbeek** [SPECFEM/specfem3d#1681] [SPECFEM3D General]
Hi all,
I have a question for any experts in attenuation using SPECFEM. It seems that 3D attenuation models may be defined by specifying Qp and Qs parameters when building the mesh. I'm wondering if it would be possible to load in more general 3D models of attenuation by reading in complex-valued elastic tensor components. For example, effective-medium models for fluid-filled aligned fractures (e.g. Chapman, 2003) predict directionally dependent attenuation. If I wanted to propagate waves using the complex elastic tensors predicted from theses effective medium models, is this something that could be done with SPECFEM? I realise that the elastic tensor in this case is frequency-dependent but for long periods this dependence is likely negligible.

Thanks for any help!
Brandon

**Comment by jpampuero** [comment]
SPECFEM solves the time-domain wave equation with real-valued parameters. AFAIK, it does not work directly with complex-valued and/or frequency-dependent elastic moduli. You first have to formulate your problem as a system of time-domain differential equations (in which frequency is not involved explicitly). For instance, an imaginary term proportional to i*frequency would be replaced by a time-derivative of strain if it's in the numerator, or of stress if it's in the denominator. You might need to introduce auxiliary variables. 
For example, see the passage from the frequency-dependent equations 8 & 10 to the time-dependent equation 12 in Komatitsch & Tromp (2003, https://doi.org/10.1046/j.1365-246X.2003.01950.x, a paper about Perfectly Matched Layers, which are basically an attenuating material initially formulated in frequency-domain).

**Reply by bvanderbeek** [reply]
Thanks for the reply and reference! Details of modelling attenuation is a new topic for me
